### ISSUE TITLE
Need a "Raku" download button/link for US Patent on Raku

### ISSUE THREAD

**Issue by autarch** [Raku/problem-solving#193]
TPF has filed for a trademark on "Raku" in the US. There were a couple objections raised by the trademark office (see https://tsdr.uspto.gov/documentviewer?caseId=sn88782119&docId=OOA20200427#docIndex=1&page=1&tdrlink= for the letter). 

One of the issues is that the raku website does not show the "raku software" available for download, only "rakudo". So in order to get the application approved we will need a download link or button that says "Raku" somewhere.

**Comment by vrurg** [comment]
Do you know if it has to be software? I mean, we could point the link to rakudo, but it'd be close to cheating. A link to spectests would be more appropriate.

Alternatively, would it be ok if the link would point to a page with a list of implementations to download? Event though it'd be just rakudo on that page.

**Comment by autarch** [comment]
> Do you know if it has to be software? I mean, we could point the link to rakudo, but it'd be close to cheating. A link to spectests would be more appropriate.

So the patent is for the name "Raku" in the area of "downloadable software". So yes, I think there needs to be something where you can "download Raku". And you might suggest a different area for the patent, but I spent a long time looking through the options and A) this is the best one; B) what we did for "Perl".

That said, I _think_ that a link that points to Rakudo would be okay.

> Alternatively, would it be ok if the link would point to a page with a list of implementations to download? Event though it'd be just rakudo on that page.

Great question, I have no idea. But I think we could try that and see what they say.

**Comment by pmichaud** [comment]
> So the patent is for the name "Raku" in the area of "downloadable software". So yes, I think there needs to be something where you can "download Raku". And you might suggest a different area for the patent, but I spent a long time looking through the options and A) this is the best one; B) what we did for "Perl".

I think the above needs   s/patent/trademark/g   , since the two are not at all the same.

Alternatively, if "patent" is indeed what is intended/contemplated, then I'm very confused.

Pm

**Comment by AlexDaniel** [comment]
> Alternatively, if "patent" is indeed what is intended/contemplated, then I'm very confused.

Nah, it seems like the usual confusion (trademarks vs patents vs copyright vs designs vs contracts). It seems to be a Trademark Application. I have changed the title.

**Comment by AlexDaniel** [comment]
I do not think that this is an infrastructure issue. The first question we should be asking is whether “Raku” should be trademarked at all. It is actually right that trademarks can not apply to the name of the language itself but rather the compiler. Didn't they mean to trademark “Rakudo”? And if they wanted to trademark “Raku”, then why?

Also, as I'm reading this:

> A display specimen for downloadable software (1) must show use of the mark directly associated with the goods and (2) such use must be of a point-of-sale nature.  37 C.F.R. §2.56(b)(1).  To show use of a point-of-sale nature, a specimen generally must provide sufficient information to enable the user to download or purchase the software from a website.  See TMEP §904.03(a) (citing In re Azteca Sys., Inc., 102 USPQ2d 1955, 1957 (TTAB 2012)). 

> In this case, the specimen does not provide the means to enable the user to download or purchase the software from the website.  See In re Sones, 590 F.3d 1282, 1286-89, 93 USPQ2d 1118, 1122-24 (Fed. Cir. 2009); In re Azteca Sys., Inc., 102 USPQ2d at 1957; TMEP §904.03(e), (i).  Specifically, the only reference to downloading a program states that it is “RAKUDO” rather than “RAKU”. 

I'm starting to wonder what kind of protections a Raku trademark will impose. For example, if I start new a Raku compiler, then create a website that has a user-friendly button saying “Download Raku now!”, will I be in violation? If yes, then I don't think it's right. If no, then what kind of protections is Raku trademark expected to bring?

**Comment by lizmat** [comment]
I'm sorry to see that TPF did not realize the difference between Raku and Rakudo.

I'm not sure that one can create a trademark for a programming language.  How does Python do this n with all of its alternate implementations?  How is Python protected?  Is PyPy trademarked?

**Comment by autarch** [comment]
> I'm starting to wonder what kind of protections a Raku trademark will impose. For example, if I start new a Raku compiler, then create a website that has a user-friendly button saying “Download Raku now!”, will I be in violation? If yes, then I don't think it's right. If no, then what kind of protections is Raku trademark expected to bring?

The trademark imposes no particular restrictions other than what the trademark holder decides to impose. TPF would defer to the wishes of the Raku community in these matters.

The main benefit of the trademark is that it prevents someone from using the name and logo in a way that conflicts with community goals. For example, if someone were to make a "Raku compiler" that was actually a proprietary word processor, that would be a case where having the trademark would be useful. Without a trademark, there's literally nothing that could be done about this.

I think the Python Foundation's statement of "General Goals" at https://www.python.org/psf/trademarks/ is a good summary of how a language name trademark should be used.

> How is Python protected?

With a trademark. See https://www.python.org/psf/trademarks/

From what I can see, PyPy is not trademarked, nor is Jython. That said, they _could_ be.

> I'm sorry to see that TPF did not realize the difference between Raku and Rakudo.

I'm not sure why you would assume this it the case. Trademarking Raku makes sense in the same way as we have a trademark for "Perl", and the same way the Python Foundation trademarked "Python".

We could also trademark "Rakudo", though I don't think this would have as much value.

**Comment by tmtvl** [comment]
I don't think the comparison to Python is as apt as a comparison to Lisp would be. That said, I don't know the legalities surrounding Lisp, so I'm afraid I can't be of much help.

**Comment by lizmat** [comment]
> the difference between Raku and Rakudo

Python is not a language definition, but an implementation.  Therefore it can have a download button.

Perl is not a language definition, but an implementation.  Therefore it can have a download button.

Rakudo is not a language definition, but an implementation.  Therefore it can have a download button.

Raku is a language definition.  I guess we could use a button to download roast?

**Comment by vrurg** [comment]
@lizmat As it was said, a link to a list of available compilers could be fine.

**Comment by jnthn** [comment]
@autarch Thanks to you and TPF for looking in to this.

I share the concerns expressed here on the language specification vs. language implementation difference, and think it's important the trademark is obtained on a solid foundation (it'd be a pity to get it, only to find this GitHub issue with all of its doubts shows up as a court exhibit later... :-))

The situation we have seems to be to be closer to, for example, the [ECMAScript trademark](https://www.ecma-international.org/memento/Ecma%20trademark%20matters.html), which is also a trademark relating to a language standard rather than an implementation. (Our standard is executable, but I doubt that makes a difference here.) Is it possible to find out the "basis" for that trademark? Perhaps they also do it as downloadable software, in which case I'd be quite reassured.

**Comment by autarch** [comment]
@jnthn Yes, here's the ECMAScript patent - http://tmsearch.uspto.gov/bin/showfield?f=doc&state=4804:ik71uw.2.1

And yes, it looks like theirs is in a different are. I think it may be possible to amend our application to use the same areas as theirs, which would make sense. I'll leave this issue open for comments for a bit to see if anyone objects or has other ideas.

**Comment by autarch** [comment]
Here's a link to the ECMAScript trademark which (I hope) will not expire - https://tsdr.uspto.gov/#caseNumber=85383290&caseType=SERIAL_NO&searchType=statusSearch

**Comment by autarch** [comment]
So I'm wondering if we should consider trademarking _both_ Rakudo and Raku. The Rakudo name would be for downloadable software, and Raku as a language spec (using the same categories as ECMAScript).

I'll bring this up at the TPF board meeting this Friday, but I'd love some feedback from the Raku community on this too.

**Comment by pmichaud** [comment]
From my perspective, it's worth trademarking both.

Pm

**Comment by vrurg** [comment]
My vote for both too.

**Comment by jnthn** [comment]
Sorry, somehow forgot to reply, but I'd be in favor of the two also.

**Comment by duncand** [comment]
Having my own experience with trademarks in this area (I trademarked "Muldis"), I believe its perfectly reasonable to trademark "Raku", but it may be good to broaden it a bit to cover technical documentation, software, and services, rather than being so specific as "downloadable software".  I also agree that a separate trademark for "Rakudo" should be gotten.

**Comment by autarch** [comment]
Hi all,

I've started with working an IP lawyer to do this properly. Here's what he suggested ...

> A search of the USPTO acceptable goods database didn't retrieve any "computer documentation" entries.  The closest entry I found is "downloadable documents in the field of .... provided via a website"  Perhaps, "downloadable documents in the field of computer programming languages provided via a website"?  

He suggests that we can use a screenshot of [docs.raku.org](https://docs.raku.org/) as support for this. This makes sense to me. Do others following this issue agree?

> As to Rakudo, how about adding a clarification that it implements the programming language to the specification used in the current Raku application, e.g. "Downloadable computer software for implementing an object-oriented, procedural, or functional programming language; downloadable computer software for use in developing, analyzing, coding, checking and controlling other computer software; downloadable computer software for use in cross-platform software application, software component, and web application development."

For this he suggests we use [the rakudo.org home page](https://rakudo.org/) as our supporting screenshot.

I think this also makes sense, but I had a couple additional questions for the Raku community.

I assume that we want to use Camelia as the trademarked logo for Raku, but please correct me if that's not correct.

Do we want to use the logo on rakudo.org in the trademark application for Rakudo? I think the alternative is to simply not trademark _any_ logo.

**Comment by vrurg** [comment]
My opinion would be "yes" to all three.

**Comment by autarch** [comment]
Okay, further clarification. We are not trademarking the logos right now (we could do that later). Right now we're just trademarking the _names_, so I think I have everything I need to go forward.

**Comment by vrurg** [comment]
My apologies, but I wasn't clear enough. I would register the logo. "Yes" meant to be "register everything". My point is that even if ever decide to modify the logo I'd protect the current one for historical reasons.

**Comment by autarch** [comment]
@vrurg No, you were clear. I was the one who was confused. I realized my earlier questions about logos weren't relevant to the trademarks we're filing for. If we wanted to trademark the logos that'd be a separate filing. It also occurs to me that since Larry drew Camelia (IIRC) that we should really have his blessing before we trademark that anyway.

**Comment by codesections** [comment]
Here's an update/summary of where we are with the Raku/Rakudo trademarks as of Sept. 24:

Our long-term plan is to have The Perl Foundation apply for three separate marks:

1. A trademark on the word "Raku"
2. An image trademark on the Camilla logo
3. A trademark on the word "Rakudo"

Of these, we have only started working with an attorney for (1).  The blocker on moving forward is figuring out the specific categories to apply for – we obviously want the broadest protection possible without risking invalidating the trademark.  Prompted by @jnthn's comment upthread, we are currently planning to follow the example set by ECMAScript which – like Raku – is a software standard rather than an implementation.  The [ECMAScript trademark](https://tsdr.uspto.gov/#caseNumber=85383290&caseType=SERIAL_NO&searchType=statusSearch) covers two classes: `009` (computer software) and `042` (design and development of computer software); we plan ask the attorney to seek both of these categories for Raku.  It is possible that we may need to provide additional evidence to support the `009` classification, but we should be able to do so.

On (2), we need to check that no one has already trademarked Camilla.  The [text description](https://github.com/Raku/mu/blob/master/misc/camelia.txt) of the Camilla logo states that Larry Wall holds both the copyright _and_ trademark rights to Camilla.  However, I've searched the USPTO trademark database and haven't found any indication that Camilla was registered (though I could have missed something).  We've reached out to @TimToady to confirm the details.

On (3), I don't believe there's any real blocker, though it's possible we may need to as TPM's board for a bit more money (~$500) to cover the expense of registering an additional trademark.

**Comment by codesections** [comment]
Av

**Comment by codesections** [comment]
After discussing the issue with the attorney we're working with, I believe we need take the following two steps to secure our trademark to "Raku".  We need to act fairly promptly because our amended trademark application is due later this month.

###  1) Add a reference to Raku to the message displayed when Rakudo is launched

The attorney asked that the message "set the marks apart from the text by vertical separation and font differences and to include the TM symbol after each".  They suggested a longer, four-line version, but we plan to suggest the following two-line compromise:
> Welcome to Rakudo™ v2020.10
implementing the Raku™ Programming Language v6.d

### 2) Supply a translation for "Raku"

Because "Raku" is a word in Japanese, the Trademark Office requires us to provide a translation of the word, along with evidence of the translation or to agree to the translation that the Trademark Office found.  They found the translation "easy" from [Google Translate](https://translate.google.com/#view=home&op=translate&sl=ja&tl=en&text=raku).  However, my understanding is that our use of "Raku" has more to do with its [meaning related to pottery](https://colabti.org/irclogger/irclogger_log/perl6?date=2018-10-25#l584).  Based on that, I think we should submit the translation "hand-moulded, lead glazed earthenware pottery fired at low temperatures".  ([citation 1](https://jisho.org/search/raku); [citation 2](https://jisho.org/search/%E6%A5%BD%E7%84%BC%E3%81%8D).

We plan to reply to the attorney in a later today, so if anyone disagrees with either of these ideas, please say so as soon as possible.  (The attorney won't file anything for a few days – and certainly not before we've updated the Rakudo message – so nothing will be final immediately but it would be better to limit the amount of back and forth over this as much as possible).

#### Other (non-essential) steps
I believe that the two steps above are all we need to do to secure our trademark.  However, there are a few other changes that might be helpful, including minor changes to the website (e.g., using ™ in a few more places and changing some of the wording around the download options) and revising the [6.d specification announcement](https://marketing.raku.org/id/1541379592/pdf_digital) to reflect the name change.  I plan to get started on PRs for these changes, but they don't need to hold up the trademark application.

(I've used the pronoun "I" in this comment because I'm speaking only for myself, but the rest of the TPF legal committee has also been involved and extremely helpful – especially Nigel Hamilton, who has been the main point of contact with the attorney and thus has taken on a large share of the work)

**Comment by lizmat** [comment]
I think Larry actually intended both meanings to apply: the first about the use of the Raku Programming Language, the second on that, even though the language will have imperfections, it will nonetheless be beautiful and thus perfect after all.

**Comment by codesections** [comment]
@lizmat thanks, that's helpful to know.  We'll ask the attorney if we can supply more than one translation.  If we can only supply one, do you have a preference about which one we supply?

**Comment by lizmat** [comment]
I would lean toward the second.  But maybe @TimToady or @thoughtstream would like to chime in at this moment.

**Comment by autarch** [comment]
I swear I remember Raku name discussions also talking about "Rakuda", which is one possible Japanese word for camel. See https://jisho.org/search/%E9%A7%B1%E9%A7%9D and https://translate.google.com/#view=home&op=translate&sl=ja&tl=en&text=%E9%A7%B1%E9%A7%9D

**Comment by codesections** [comment]
> I swear I remember Raku name discussions also talking about "Rakuda", which is one possible Japanese word for camel. 

Yeah, I came across that too – for example, it was prominently mentioned in the [HackerNews discussion around the rename](https://news.ycombinator.com/item?id=21227430).  However, the request from the Trademark Office is specifically to provide _translations_ of the term "Raku", so I don't think there's a good way for us to include statements about "Rakuda" in the trademark application. 

(As an entirely separate matter from the TM application, it might be nice to have something in our FAQ about what the name means.  We've (unsurprisingly!) got a lot of puns/linguistic shenanigans going on, and I'm not sure that new users appreciate all the layers of meaning)

**Comment by thoughtstream** [comment]
> We'll ask the attorney if we can supply more than one translation. If we can only supply one, do you have a preference about which one we supply?

___[Obligatory "I am not a lawyer", "I am not TimToady", "I am not a native speaker of Japanese" disclaimers]___

I always think of the name "Raku" as primarily signifying _"comfort and ease"_. 

Yes, it's also the name for beautifully repaired broken things, but (a) that's not a primary meaning[1] and (b) _"broken but beautiful"_ is **not** the main association I want for Raku, especially not in a public forum or a legal document.

Hence I would personally favour submitting _"comfort and ease"_ as the only translation.

I would also **not** be inclined to mention the other word-play associations with "rakuda" _("camel")_ or "rakudo" _("paradise")_
unless an actual trademark lawyer advises that it would be advantageous to do so.

[1] My understanding is that the name for this pottery repair technique originally derives from the _"comfort and ease"_ meaning. But, again, I am not an expert in the Japanese language.

**Comment by lizmat** [comment]
"Paradise" might be applicable to the "Rakudo" trademark.

With regards to Raku in the pottery context: https://en.wikipedia.org/wiki/Raku_ware (which does *not* have the string "repair" in it).

**Comment by thoughtstream** [comment]
> With regards to Raku in the pottery context: https://en.wikipedia.org/wiki/Raku_ware (which does not have the string "repair" in it).

See, not a Japanese pottery expert either. ;-)
(I was evidently conflating _raku_ with _kintsugi_.)

I would still prefer the _"comfort and ease"_ connotation over the _"hand-made and irregular"_ meaning.

Though, according to the Wikipedia entry, the correct translation in that context is _"enjoyment"_.
If that's right, then I will happily start using _"enjoyment and ease"_ as my standard translation.

**Comment by codesections** [comment]
> Hence I would personally favour submitting "comfort and ease" as the only translation.

That makes sense to me.

I will say that I like the associations of "comfort and ease" **significantly** more than those of "easy" (the translation the TM office suggested).  It might be a subtle distinction, but to me "easy" in a language context suggests "easy to learn" – which not only isn't a great fit for Raku, but [I'd argue](https://www.codesections.com/blog/raku-manifesto/) is actually a non-goal for Raku.  For some reason, "comfort and ease" doesn't have that same "learnability" connotation to me – it seems more like a description of how it feels to use the language.

**Comment by codesections** [comment]
> Though, according to the Wikipedia entry, the correct translation in that context is "enjoyment".

Hmm, interesting.  Wikipedia doesn't provide a source for that translation, and the first 4 online dictionaries I checked all had "comfort" rather than "enjoyment"  (though they might all be sourcing their translations from the same place for all I know).  There is something appealing about "enjoyment", though – I'll do a bit more digging and see what I can come up with.

**Comment by vrurg** [comment]
Just an opinion: _"enjoyment and ease"_ sounds the best to me. If it's a correct one, I'd stick to it. Otherwise _comfort_ is good enough.

**Comment by pmichaud** [comment]
In the totally "for whatever it's worth" category, "enjoyment and ease" is exactly what I'd be aiming for, if it'll pass USPTO muster.  The "enjoyment and ease" notion also reminds me of -Ofun .

Pm

**Comment by codesections** [comment]
I did a bit more digging (er, ok I got help from a friend) and located [this translation](https://jitenon.com/kanji/%E6%A5%BD), which seems to support the "enjoyment and ease" translation.

Based on this source and the discussion above, we'll try to use that translation for the trademark application. Thanks, all.

**Comment by codesections** [comment]
Update as of October 27:  the attorney representing us has now filled [a Response](https://tsdr.uspto.gov/documentviewer?caseId=sn88782119&docId=ROA20201026190511#docIndex=0&page=1) that provides the translation "enjoyment and ease" for  "raku".  The response also amends the basis for establishing the trademark, which will allow us to rely on the image of the Raku welcome screen shown below:
![image](https://user-images.githubusercontent.com/35405463/97366410-aac5cb00-187d-11eb-9458-b6ef63cdbdfe.png)

The attorney will provide the trademark office the image once the trademark file has been updated.

**Comment by autarch** [comment]
TPF is currently working on both trademarks in the US. I think this can be closed.
